qa/tasks/cephfs/test_snap_schedules.py

```diff
@@ -453,4 +454,50 @@ def test_snap_schedule_with_mgr_restart(self):
         # cleanup
         self.fs_snap_schedule_cmd('remove', path=testdir, snap_schedule='1M')
         self.remove_snapshots(testdir[1:])
-        self.mount_a.run_shell(['rmdir', testdir[1:]])
+        self.mount_a.run_shell(['rmdir', testdir[1:]])    
+
+class TestSnapSchedulesSnapdir(TestSnapSchedulesHelper):
+    def remove_snapshots(self, dir_path, sdn):
+        snap_path = f'{dir_path}/{sdn}'
+
+        snapshots = self.mount_a.ls(path=snap_path)
+        for snapshot in snapshots:
+            snapshot_path = os.path.join(snap_path, snapshot)
+            log.debug(f'removing snapshot: {snapshot_path}')
+            self.mount_a.run_shell(['rmdir', snapshot_path])
+
+    def get_snap_dir_name(self):
+        from tasks.cephfs.fuse_mount import FuseMount
+        from tasks.cephfs.kernel_mount import KernelMount
+
+        sdn = self.fs.get_config('client_snapdir')
+        if sdn == None:
+            if isinstance(self.mount_a, KernelMount):
+                sdn = self.mount_a.client_config.get('snapdirname', '.snap')
+            elif isinstance(self.mount_a, FuseMount):
+                sdn = self.mount_a.client_config.get('client_snapdir', '.snap')
+        return sdn
```

Comment:
Only with this I am afraid the teuthology test still won't work, you need to apply the options to fuse/kernel client.

For kclient you need to do what the `rbytes` does in `kernel_mount.py`.

While for fuse client it's a little complex, you need to set the value to the `ceph.conf` file first by using:

```
self.fs.set_ceph_conf('client', 'client snapdir', sdn)
```

And then remount the `mount_a`.

Then all the options you set in the `.yaml` could be applied to the fuse client daemon and kernel mount.

Or without the above the `mount_a` will still use the default `.snap`.

Reply:
For the fuse client, I am not sure whether setting the option `client_snapdir` to the mounter daemon directly will work.

Reply:
For the fuse client, I am not sure whether setting the option `client_snapdir` to the mounter daemon directly will work.

Reply:
@neesingh-rh ping? (this comment)

Reply:
@vshankar @lxbsz I had updated the code with requested changes here.Needs re-review.